cloud/cloud_env_impl.cc

```diff
@@ -1441,31 +1453,27 @@ Status CloudEnvImpl::MaybeMigrateManifestFile(const std::string& local_dbname) {
   if (!st.ok()) {
     return st;
   }
-  // Note: This rename is important for migration. If we are just starting on
-  // an old database, our local MANIFEST filename will be something like
-  // MANIFEST-00001 instead of MANIFEST. If we don't do the rename we'll
-  // download MANIFEST file from the cloud, which might not be what we want do
-  // to (especially for databases which don't have a destination bucket
-  // specified). This piece of code can be removed post-migration.
   manifest_filename = local_dbname + "/" + rtrim_if(manifest_filename, '\n');
   if (local_env->FileExists(manifest_filename).IsNotFound()) {
     // manifest doesn't exist, shrug
     Log(InfoLogLevel::INFO_LEVEL, info_log_,
-        "[cloud_env_impl] MaybeMigrateManifestFile: Manifest %s does not exist",
+        "[cloud_env_impl] MigrateFromPureRocksDB: Manifest %s does not exist",
         manifest_filename.c_str());
     return Status::OK();
   }
-  return local_env->RenameFile(manifest_filename, local_dbname + "/MANIFEST");
+  st = local_env->RenameFile(manifest_filename, local_dbname + "/MANIFEST");
+  if (!st.ok()) {
+    return st;
+  }
+
+  return RollNewEpoch(local_dbname);
```

Comment:
Every time we open a database we need to roll new epoch. Other part of the code won't roll the new epoch because the cloud manifest has already been loaded here.